ls1intum/Artemis
# `Lecture`: Allow instructors to remove solution slides when processing units

### Checklist
#### General


#### Server
#### Client


### Motivation and Context


Instructors are not able to remove "Example solution" slides when processing units automatically.

### Description


A new checkbox is provided to choose if the instructors want to remove the solution slides. Then all example solution slides that contain "Example solution", "Example solution: model" or "Example solution (with comments)" will be removed.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course

1. Log in to Artemis
2. Navigate to Course Administration
3. Create a new Lecture and choose to automatically process the file.
4. Then you will see a table with split information and at the bottom you will have the option to choose to remove Break or Solution slides. 
5. Select both and then submit the form. 
6. See the file units if the slides are removed correctly.
example pdf: 

[ITP2223 L05 Object Orientation I_organized.pdf](https://github.com/ls1intum/Artemis/files/12096274/ITP2223.L05.Object.Orientation.I_organized.pdf)



### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage








#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [attachment-units.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5172/latest/artifact/shared/Coverage-Report-Client-Tests/app/lecture/lecture-unit/lecture-unit-management/attachment-units/attachment-units.component.ts.html) | 89.47% | ✅ |

#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [LectureUnitProcessingService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5172/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service/LectureUnitProcessingService.html) | 97% | ✅ |
| [LectureUnitInformationDTO.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5172/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest.dto/LectureUnitInformationDTO.html) | 100% | ✅ |

�

### Screenshots



New checkbox introduced:
![Screenshot 2023-07-13 at 18 10 08](https://github.com/ls1intum/Artemis/assets/26312757/e03702e2-172a-487f-9892-0dbe8f4ee558)
